integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+func TestAtlasPlannedMaintenance(t *testing.T) {
+	connstring := os.Args[1]
+	workloadSpec := []byte(os.Args[2])
+
+	setupOpts := mtest.NewSetupOptions().SetURI(connstring)
+	if err := mtest.Setup(setupOpts); err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := mtest.Teardown(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// killAllSessions will return an auth error if it's run
+	fileReqs, runners := unified.ParseTestFile(t, workloadSpec, unified.NewOptions().SetRunKillAllSessions(false))
+
+	mtOpts := mtest.NewOptions().
+		RunOn(fileReqs...).
+		CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	// a testfile can parse to multiple runners, though we only expect to get one per file
+	for _, runner := range runners {
```

Comment:
How would a test file with multiple test cases work? If more than one test case had a `loop` operation, we'd be expecting multiple SIGTERM signals because each test case would block indefinitely. I don't see anything in the executor spec about this, so we should at least get that clarified in the document.